src/frontend/src/pages/TeamsPage/TeamSpecificPage.tsx

```diff
@@ -13,53 +13,117 @@ import { Delete } from '@mui/icons-material';
 import { NERButton } from '../../components/NERButton';
 import { useCurrentUser } from '../../hooks/users.hooks';
 import { isAdmin } from 'shared';
-import { useState } from 'react';
+import React, { useState } from 'react';
 import DeleteTeamModal from './DeleteTeamModal';
 import SetTeamTypeModal from './SetTeamTypeModal';
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
+import { TeamPill } from './TeamPill';
+import { useToast } from '../../hooks/toasts.hooks';
 
 interface ParamTypes {
   teamId: string;
 }
 
 const TeamSpecificPage: React.FC = () => {
+  const toast = useToast();
   const { teamId } = useParams<ParamTypes>();
   const { isLoading, isError, data, error } = useSingleTeam(teamId);
   const user = useCurrentUser();
   const [showDeleteModal, setShowDeleteModal] = useState(false);
   const [showTeamTypeModal, setShowTeamTypeModal] = useState(false);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const dropdownOpen = Boolean(anchorEl);
+  const { mutateAsync: archiveTeam } = useArchiveTeam(teamId);
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleDropdownClose = () => {
+    setAnchorEl(null);
+  };
 
   if (isError) return <ErrorPage message={error?.message} />;
   if (isLoading || !data) return <LoadingIndicator />;
 
   const DeleteButton = () => (
     <NERButton
       variant="contained"
-      disabled={!isAdmin(user.role)}
       startIcon={<Delete />}
       onClick={() => setShowDeleteModal(true)}
-      sx={{ marginRight: '10px' }}
+      disabled={!isAdmin(user.role)}
     >
       Delete
     </NERButton>
   );
 
   const SetTeamTypeButton = () => (
-    <NERButton variant="contained" disabled={!isAdmin(user.role)} onClick={() => setShowTeamTypeModal(true)}>
+    <NERButton variant="contained" onClick={() => setShowTeamTypeModal(true)} disabled={!isAdmin(user.role)}>
       Set Team Type
     </NERButton>
   );
 
+  interface ArchiveTeamButtonProps {
+    archive: boolean;
+  }
+
+  const handleArchive = async () => {
+    try {
+      await archiveTeam(teamId);
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        toast.error(e.message, 3000);
+      }
+    }
+  };
+
+  const ArchiveTeamButton: React.FC<ArchiveTeamButtonProps> = ({ archive }) => (
+    <NERButton variant="contained" onClick={() => handleArchive()} disabled={!isAdmin(user.role)}>
+      {archive ? 'Unarchive Team' : 'Archive Team'}
+    </NERButton>
+  );
+
+  const TeamActionsDropdown = () => (
```

Comment:
You don't have to use buttons in the dropdown itself - you can model the dropdown after the project actions dropdown to see how we typically do it